lib/chai/core/assertions.js

```diff
@@ -1532,6 +1532,39 @@ module.exports = function (chai, _) {
   });
 
   /**
+   * ### .oneOf(list)
+   *
+   * Assert that a non-array, non-object value appears somewhere in the flat array `list`.
+   *
+   *     expect('a').to.be.oneOf(['a', 'b', 'c']);
+   *     expect(9).to.not.be.oneOf(['z']);
+   *
+   * @name oneOf
+   * @param {Array<*>} list
+   * @param {String} message _optional_
+   * @api public
+   */
+
+  function oneOf (list, msg) {
+    if (msg) flag(this, 'message', msg);
+    var expected = flag(this, 'object');
+    new Assertion(list).to.be.an('array');
+    new Assertion(expected).to.not.be.an('array');
+    new Assertion(expected).to.not.be.an('object');
```

Comment:
I could probably get that in there. Just so you know, it'll act like Python's `in` keyword. Meaning:

```py
>>> [3] in [1, 2, [3]]
True
>>> [4] in [1, 2, [3, [4]]]
False
>>> [3, [4]] in [1, 2, [3, [4]]]
True
```

In other words, unless I'm matching something from the *first level* all the way to the end of that first level object-type, it'll fail. That's the simplest way I can think to handle object-types inside of a list without a `deep` flag.